src/custom-element.js

```diff
@@ -343,1087 +343,1100 @@ function isInternalOrServiceNode(node) {
+       * element that will be used to size this element.
+       * @private {?Element}
+       */
+      this.sizerElement_ = null;
 
-    // `opt_implementationClass` is only used for tests.
-    let Ctor = knownElements[this.name];
-    if (getMode().test && this.implementationClassForTesting) {
-      Ctor = this.implementationClassForTesting;
-    }
+      /** @private {boolean|undefined} */
+      this.loadingDisabled_ = undefined;
 
-    /** @private {!./base-element.BaseElement} */
-    this.implementation_ = new Ctor(this);
+      /** @private {boolean|undefined} */
+      this.loadingState_ = undefined;
 
-    /**
-     * An element always starts in a unupgraded state until it's added to DOM
-     * for the first time in which case it can be upgraded immediately or wait
-     * for script download or `upgradeCallback`.
-     * @private {!UpgradeState}
-     */
-    this.upgradeState_ = UpgradeState.NOT_UPGRADED;
+      /** @private {?Element} */
+      this.loadingContainer_ = null;
+
+      /** @private {?Element} */
+      this.loadingElement_ = null;
+
+      /** @private {?Element|undefined} */
+      this.overflowElement_ = undefined;
+
+      // `opt_implementationClass` is only used for tests.
+      let Ctor = knownElements[this.name];
+      if (getMode().test && this.implementationClassForTesting) {
+        Ctor = this.implementationClassForTesting;
+      }
+      /** @private {./base-element.BaseElement} */
+      this.implementation_ = new Ctor(this);
+
+      /**
+       * An element always starts in a unupgraded state until it's added to DOM
+       * for the first time in which case it can be upgraded immediately or wait
+       * for script download or `upgradeCallback`.
+       * @private {!UpgradeState}
+       */
+      this.upgradeState_ = UpgradeState.NOT_UPGRADED;
+
+      /**
+       * Action queue is initially created and kept around until the element
+       * is ready to send actions directly to the implementation.
+       * @private {?Array<!./service/action-impl.ActionInvocation>}
+       */
+      this.actionQueue_ = [];
+
+      /**
+       * Whether the element is in the template.
+       * @private {boolean|undefined}
+       */
+      this.isInTemplate_ = undefined;
+    }
 
     /**
-     * Action queue is initially created and kept around until the element
-     * is ready to send actions directly to the implementation.
-     * @private {?Array<!./service/action-impl.ActionInvocation>}
+     * The Custom Elements V0 equivalent to the constructor.
+     * @suppress {checkTypes}
      */
-    this.actionQueue_ = [];
+    createdCallback() {
+      this.init_();
+    }
 
     /**
-     * Whether the element is in the template.
-     * @private {boolean|undefined}
+     * Returns the associated ampdoc. Only available after attachment. It throws
+     * exception before the element is attached.
+     * @return {!./service/ampdoc-impl.AmpDoc}
+     * @final @this {!Element}
```

Comment:
Reverted this change since `@this` is how the old code avoided these same type errors. I think this is slightly better replacing them with `@suppress {missingProperties}`.